File: source/rendering/lodResult.cpp
/*
 * This file is part of Cosmonium.
 *
 * Copyright (C) 2018-2021 Laurent Deru.
 *
 * Cosmonium is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Cosmonium is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Cosmonium.  If not, see <http://www.gnu.org/licenses/>.
 */

#include "quadTreeNode.h"
#include "lodResult.h"

LodResult::LodResult(void) :
  to_split(),
  to_merge(),
  to_show(),
  to_remove(),
  max_lod(0)
{
}

LodResult::~LodResult(void)
{
}

void
LodResult::add_to_split(QuadTreeNode *patch)
{
  to_split.add(patch);
}

void
LodResult::add_to_merge(QuadTreeNode *patch)
{
  to_merge.add(patch);
}

void
LodResult::add_to_show(QuadTreeNode *patch)
{
  to_show.add(patch);
}

void
LodResult::add_to_remove(QuadTreeNode *patch)
{
  to_remove.add(patch);
}

void
LodResult::check_max_lod(QuadTreeNode *patch)
{
  max_lod = max(max_lod, patch->lod);
}

void
LodResult::sort_by_distance(void)
{
  to_split.sort_by_distance();
  to_merge.sort_by_distance();
  to_show.sort_by_distance();
  to_remove.sort_by_distance();
}

File: cosmonium/pyrendering/lodresult.py
#
# This file is part of Cosmonium.
#
# Copyright (C) 2018-2024 Laurent Deru.
#
# Cosmonium is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Cosmonium is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Cosmonium.  If not, see <https://www.gnu.org/licenses/>.
#


class LodResult:
    def __init__(self):
        self.to_split = []
        self.to_merge = []
        self.to_show = []
        self.to_remove = []
        self.max_lod = 0

    def add_to_split(self, patch):
        self.to_split.append(patch)

    def add_to_merge(self, patch):
        self.to_merge.append(patch)

    def add_to_show(self, patch):
        self.to_show.append(patch)

    def add_to_remove(self, patch):
        self.to_remove.append(patch)

    def check_max_lod(self, patch):
        self.max_lod = max(self.max_lod, patch.lod)

    def sort_by_distance(self):
        self.to_split.sort(key=lambda x: x.distance)
        self.to_merge.sort(key=lambda x: x.distance)
        self.to_show.sort(key=lambda x: x.distance)
        self.to_remove.sort(key=lambda x: x.distance)
